efi/fw_load_handler.go

```diff
@@ -20,9 +20,251 @@
 package efi
 
 import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	efi "github.com/canonical/go-efilib"
+	"github.com/canonical/go-tpm2"
 	"github.com/canonical/tcglog-parser"
+	"golang.org/x/xerrors"
+)
+
+const (
+	sbStateName = "SecureBoot" // Unicode variable name for the EFI secure boot configuration (enabled/disabled)
 )
 
+// fwLoadHandler is an implementation of imageLoadHandler that measures firmware
+// events (pre-OS events and events related to loading of OS components).
+type fwLoadHandler struct {
+	log *tcglog.Log
+}
+
 var newFwLoadHandler = func(log *tcglog.Log) imageLoadHandler {
-	panic("not implemented")
+	return &fwLoadHandler{log: log}
+}
+
+func (h *fwLoadHandler) measureSignatureDb(ctx pcrBranchContext, name efi.VariableDescriptor) ([]byte, error) {
```

Comment:
does this need a slightly different name as it also returns the db, most measure* methods return only errors, but some return more

Reply:
I've renamed this to `readAndMeasureSignatureDb`